lib/location/method_gnss.c

```diff
@@ -1175,38 +1142,18 @@ static void method_gnss_agnss_expiry_process(const struct nrf_modem_gnss_agnss_e
 	}
 }
 
-/* Queries assistance data need from GNSS.
- *
- * To maintain backward compatibility with older modem firmware versions, two different methods are
- * supported. If the nrf_modem_gnss_agnss_expiry_get() API is not supported (pre-v1.3.2 modem
- * firmware), GNSS is briefly started to trigger the NRF_MODEM_GNSS_EVT_AGNSS_REQ event from GNSS
- * in case assistance data is currently needed.
- */
+/* Queries assistance data need from GNSS. */
 static void method_gnss_agnss_req_work_fn(struct k_work *work)
```

Comment:
I'm thinking if this still needs to be running as a separate worker... Not sure but now that start and stop are not used, are all functions simply synchronous. Things would simplify and we might get rid of a sleep on line 1264.

Maybe function name could be something like `method_gnss_agnss_data_needed()` etc.

Reply:
I think the extra sleep could be removed without making the function synchronous as well, but you are right, this can now be simplified further by making the A-GNSS data need check synchronous. I'll change it.